spec/syntax.md

```diff
@@ -53,6 +64,122 @@ No whitespace is allowed between components of an `InterpolatedIdentifier`.
 
 ## Procedures
 
+### Parsing Text as CSS
+
+This algorithm takes a string, `text`, and returns a Sass abstract syntax tree.
+
+> This algorithm is designed with two goals in mind:
+>
+> 1. CSS imported from Sass should be as compatible with standard CSS as
+>    possible. In some cases we err even more towards CSS compatibility than
+>    SCSS does, because the CSS being imported is likely not written by someone
+>    who knows to avoid things that Sass interprets specially (such as certain
+>    `@import` URLs).
+>
+> 2. We should provide clear and eager feedback to users who accidentally try to
+>    use Sass features in CSS imports. We don't allow these features, and we
+>    want users to know that through error messages rather than digging through
+>    generated CSS only to find that Sass features were passed through
+>    unmodified. This is a particular concern because LibSass has historically
+>    allowed the use of Sass features in CSS imports.
+
+The algorithm for parsing text as CSS works like parsing text as SCSS, with some
+modifications. The following productions should produce errors:
+
+* Any at-rules that are defined in Sass and not in plain CSS. At the time of
+  writing, this means:
+
+  * `@at-root`
+  * `@content`
+  * `@debug`
+  * `@each`
+  * `@error`
+  * `@extend`
+  * `@for`
+  * `@function`
+  * `@if`
+  * `@include`
+  * `@mixin`
+  * `@return`
+  * `@warn`
+  * `@while`
+
+* An `@import` that contains interpolation in the `url()`, the media query, or
+  the supports query.
+
+* An `@import` that appears within a style rule or at-rule.
+
+* An `@import` with more than one argument.
+
+* A declaration followed by an open curly brace (that is, a nested declaration).
+
+* A style rule appearing within another style rule.
+
+* The parent selector `&`, either in a selector or a declaration value.
+
+* Placeholder selectors.
+
+* All built-in functions, *excluding* the following:
+
+  * `rgb()`
+  * `rgba()`
+  * `hsl()`
+  * `hsla()`
+  * `grayscale()`
+  * `invert()`
+  * `alpha()`
+  * `opacity()`
```

Comment:
This is intended to specifically refer to constructs that would call a built-in function definition, so `min()` and `max()` used in a plain-CSS manner are okay because they're parsed differently. `calc()`, `env()`, and `var()` have no built-in function equivalents.